syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "farm_ng_proto/tractor/v1/resource.proto";
import "farm_ng_proto/tractor/v1/calibrator.proto";

package farm_ng_proto.tractor.v1;
option go_package = "github.com/farm_ng/genproto";

message CalibrateMultiViewApriltagRigConfiguration {
  // Serialized CaptureVideoDatasetResult
  Resource video_dataset = 1;
  // A list of tag ids to consider as one rigid object or "rig".
  repeated int32 tag_ids = 2;
  // This tag id will be considered the root coordinate system of the
  // rig, and all transforms will be estimated with respect to this tag.
  int32 root_tag_id = 3;
  string root_camera_name = 4;

  // name of the result.
  string name = 5;
  string tag_rig_name = 6;
  bool filter_stable_tags = 7;

}

message CalibrateMultiViewApriltagRigStatus {
  oneof input_required {
    CalibrateMultiViewApriltagRigConfiguration input_required_configuration = 1;
  }
  // Serialized CalibrateMultiViewApriltagRigResult
  Resource result = 2;

  SolverIterationSummary iteration_summary = 3;
}

message CalibrateMultiViewApriltagRigResult {
  // Configuration which initiated the computation of this result.
  CalibrateMultiViewApriltagRigConfiguration configuration = 1;
  Resource multi_view_apriltag_rig_initial = 2;
  Resource multi_view_apriltag_rig_solved = 3;
  SolverStatus solver_status = 4;
  double rmse = 5;
  google.protobuf.Timestamp stamp_end = 6;
  // Debug event log.
  Resource event_log = 7;
}
